api/src/main/java/org/apache/iceberg/expressions/UnboundPredicate.java

```diff
@@ -109,6 +109,9 @@ public Expression bind(StructType struct, boolean caseSensitive) {
       return bindUnaryOperation(bound);
     }
 
+    Preconditions.checkState(!bound.type().isNestedType(),
+        "Non-Unary Predicate is not supported for top column of nested types");
```

Comment:
Sentence is a bit complex, maybe 'Non-unary predicate is not supported for complex types'?  Or even 'Only unary predicates are supported for complex types'.